layer/pandas_extensions.py

```diff
@@ -346,3 +346,21 @@ def _register_type_extensions() -> None:
     # register pandas extension types
     register_extension_dtype(_ImageDtype)
     register_extension_dtype(_ArrayDtype)
+
+
+def _infer_custom_types(data: pd.DataFrame) -> pd.DataFrame:
```

Comment:
nit: why the prefix `_`?

Reply:
It is internal function, should not be used outside `layer`.

Reply:
It wouldn't be? as we don't export it in https://github.com/layerai/sdk/blob/main/layer/__init__.py

Reply:
Yes, but is still private, and `_` emphasises it should stay like that.